### ISSUE TITLE
`compose.yaml` `name` breaks when using vars `${VAR}`

### ISSUE THREAD

**Issue by lucasfcnunes** [devcontainers/cli#839]
# Description

#### Setup

```compose.yaml
# compose.yaml
name: ${COMPOSE_PROJECT_NAME}
services:
  # ...
```

```.env
# .env
COMPOSE_PROJECT_NAME=abc_123
```

#### Results

```sh
$ docker compose config
name: abc_123
services:
  # ...
```

#### Issue

Instead of doing the expansion as above, devcontainer now reads it as `compose_project_name` - not expanding the variable. (`${COMPOSE_PROJECT_NAME}` -> `compose_project_name`)

**It breaks the names of the project, volumes, networks, etc - empty container**

It happened first today on https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers @ `v0.369.0`.

The `v0.362.0` uses the folder name. My var is always the same as the folder name that's why I only experienced it today.

# Related

- #817
- #831
- #833

**Comment by lucasfcnunes** [comment]
@chrmarti can you help me?

**Comment by chrmarti** [comment]
Preparing a fix. You can use the previous stable version of Dev Containers 0.362.0 until we get the fix out.

**Comment by chrmarti** [comment]
This is fixed in Dev Containers 0.372.0-pre-release. Could you give that a try and let me know if it fixes the issue for you? Thanks.
